python/dgl/dataloading/base.py

```diff
@@ -271,9 +410,171 @@ def sample(self, g, seed_edges):
             g, seed_edges, exclude, self.reverse_eids, self.reverse_etypes,
             self.output_device)
 
-        input_nodes, _, blocks = self.block_sampler.sample_blocks(g, seed_nodes, exclude_eids)
+        input_nodes, _, blocks = self.sampler.sample(g, seed_nodes, exclude_eids)
 
         if self.negative_sampler is None:
             return self.assign_lazy_features((input_nodes, pair_graph, blocks))
         else:
             return self.assign_lazy_features((input_nodes, pair_graph, neg_graph, blocks))
+
+def as_edge_prediction_sampler(
+        sampler, exclude=None, reverse_eids=None, reverse_etypes=None, negative_sampler=None,
+        prefetch_labels=None):
+    """Wrap a sampler for node classification/regression into another sampler for
+    edge classification/regression or link prediction.
+
+    For each iteration, the sampler will yield
+
+    * A tensor of input nodes necessary for computing the representation on edges, or
+      a dictionary of node type names and such tensors.
+
+    * A subgraph that contains only the edges in the minibatch and their incident nodes.
+      Note that the graph has an identical metagraph with the original graph.
+
+    * If a negative sampler is given, another graph that contains the "negative edges",
+      connecting the source and destination nodes yielded from the given negative sampler.
+
+    * A list of MFGs necessary for computing the representation of the incident nodes
+      of the edges in the minibatch.
+
+    Parameters
+    ----------
+    sampler : Sampler
+        The sampler object.  It additionally requires that the :attr:`sample` method
+        must have an optional third argument :attr:`exclude_eids` representing the
```

Comment:
should not be optional?

Reply:
It should be optional because the sampler is a node-wise sampler, meaning that when one uses it for node dataloading it will only receive two arguments.